src/torchjd/autojac/mtl_backward.py

```diff
@@ -23,12 +25,92 @@
 )
 
 
+def _get_tensors_memory_id(tensor_list: list[Tensor]) -> set[int]:
+    """
+    Based on a list of lists of tensor objects returns the unique memory adress of all
+    :param tensor_list: list containing pytorch Tensor objects
+    """
+    return set([id(tensor) for tensor in tensor_list])
```

Comment:
I am not convinced this is useful since we typically use these as keys for dicts (or counters) which therefore already uses the id.